node/src/components/contract_runtime/operations.rs

```diff
@@ -152,20 +155,21 @@ pub fn execute_finalized_block(
 
         trace!(?deploy_hash, ?result, "deploy execution result");
         // As for now a given state is expected to exist.
-        let (state_hash, execution_result) = commit_execution_results(
+        let (state_hash, execution_result, messages) = commit_execution_results(
             &scratch_state,
             metrics.clone(),
             state_root_hash,
             deploy_hash,
             result,
         )?;
-        execution_results.push((deploy_hash, deploy_header, execution_result));
+        execution_results.push((deploy_hash, deploy_header, execution_result, messages));
```

Comment:
Could this be changed into a named struct at this point?

Reply:
Changed in https://github.com/casper-network/casper-node/commit/53483e16590b67de9be9be4d9ba4e6b64d1f22ee although didn't use checked_memory_slice as it's not available yet on the feat-2.0 branch. Will do a follow-up commit after.